### ISSUE TITLE
Missing semicolon erroneously raised

### ISSUE THREAD

**Issue by SirCmpwn** [eslint/eslint#7262]
**Tell us about your environment**

* **ESLint Version:**

    "babel-eslint": "^6.1.2",
    "eslint": "^2.13.1",
    "eslint-config-airbnb": "^9.0.1",
    "eslint-plugin-babel": "^3.3.0",
    "eslint-plugin-import": "*",
    "eslint-plugin-jsx-a11y": "^1.2.2",
    "eslint-plugin-react": "^5.1.1",

* **Node Version:** 6.6.0
* **npm Version:** 3.10.8

**Please show your full configuration:**

```
{
  "extends": "airbnb",
  "plugins": [
    "babel"
  ],
  "env": {
    "browser": true,
    "mocha": true
  },
  "parser": "babel-eslint",
  rules: {
    "babel/generator-star-spacing": 2,
    "babel/new-cap": 2,
    "babel/array-bracket-spacing": 2,
    "babel/object-shorthand": 2,
    "babel/flow-object-type": 2,
    "babel/no-await-in-loop": 0,
    "babel/arrow-parens": 0,
    "import/no-unresolved": 0,
    "no-underscore-dangle": 0,
    "no-unused-expressions": 0,
    "no-confusing-arrow": 0,
    "array-callback-return": 0,
    "consistent-return": 0,
    "react/no-multi-comp": 0,
    "react/no-did-mount-set-state": 0,
    "import/no-duplicates": 0,
    "no-duplicate-imports": 0,
    "no-shadow": 0,
    "radix": 0,
    "dot-notation": [
      "error", {
        "allowPattern": "^(default|public|private|return)$"
      }
    ]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
export function parallel(thunks) {
  return async (dispatch, getState) =>
    await Promise.all(thunks.map(t => t(dispatch, getState)));
}
```

**What did you expect to happen?**

No errors raised.

**What actually happened? Please include the actual, raw output from ESLint.**

      5:31  error  Missing semicolon  semi

**Comment by SirCmpwn** [comment]
Whoops, my bad, I totally misread the file that was causing the problem. Embarrassing...
